project/test_cases.py

```diff
@@ -3,46 +3,130 @@
-from project_class import Project_class
+from database_setup import Project, Base, Student, Pref
+from algorithm_16pref import get_raw_score, get_popularity_score
+from algorithm_16pref import raw_sort, get_underfilled_groups
+from algorithm_16pref import pop_sort, give_all_prefs
+from algorithm_16pref import give_first_prefs
+from algorithm_16pref import MAX_STUDS_PER_GROUP
+from test_application import create_user, get_user_by_email
+# from project_class import Project_class
 
 APP = Flask(__name__)
-ENGINE = create_engine('sqlite:///testing.db')
-Base.metadata.bind = engine
+ENGINE = create_engine('sqlite:///testdatabase.db')
+Base.metadata.bind = ENGINE
 DBSESSION = sessionmaker(bind=ENGINE)
 SESSION = DBSESSION()
 
 
 class BasicTests(unittest.TestCase):
     """Umbrella class of all test cases"""
 
-    def test_raw_score(self):
-        """Test if get raw score function works"""
-        list_score = []
-        projects = SESSION.query(Project).all()
+    def test_max_pop(self):
+        """Test to see if any population is above 15"""
+        list_num_fail = 0
+        projects  = SESSION.query(Project).all()
         for proj in projects:
-            score = get_raw_score(proj)
-            list_score.append(score)
-        expected_results = [0]
-        print "raw score: "
-        print list_score
-        self.assertEqual(list_score, expected_results)
-
-    def test_pop_score(self):
-        """Test if get pop score function works"""
-        list_score = []
-        projects = SESSION.query(Project).all()
-        for proj in projects:
-            score = get_popularity_score(proj)
-            list_score.append(score)
-        expected_results = [0]
-        print "pop score: "
-        print list_score
-        self.assertEqual(list_score, expected_results)
+            num_students = 0
+            num_students = len(proj.students)
+            if num_students >= MAX_STUDS_PER_GROUP:
+                list_num_fail += 1
+        if list_num_fail > 0:
+            print "Number of Overloaded Projects: " 
+            print list_num_fail
+            assert False, "Number of overloaded projects: " + str(list_num_fail)
+
+    def test_all_students_assigned(self):
+        """Test to see if all students are ssigned to a project"""
+        students = SESSION.query(Student).all()
+        for student in students:
+            assigned_1 = student.session_1_matched
+            assigned_2 = student.session_2_matched
+            assigned_3 = student.session_3_matched
+            assigned_4 = student.session_4_matched
+
+            if assigned_1 == False:
+                assert False
+            elif assigned_2 == False:
+                assert False
+            elif assigned_3 == False:
+                assert False
+            elif assigned_4 == False:
+                assert False
+            else: 
+                print "Students Assigned"
+    
+    def test_is_student_in_session(self):
```

Comment:
I'm still confused as to why this method is structured the way it is. 

What you should do is grab the students. Go through the projects in a session and as you see a student remove them from the students list. By the end, there should be no students in the list. Then you just rerun that method for each of the four sessions.